src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -2471,8 +2474,20 @@ public async Task SourceGenerator_DoesNotUpdateSources_WhenSourceGeneratorIsSupp
             var compilation = await project.GetCompilationAsync();
             var (driver, additionalTexts, optionsProvider) = await GetDriverWithAdditionalTextAndProviderAsync(project);
 
-            var result = RunGenerator(compilation!, ref driver);
-            result.VerifyPageOutput(
+            // start with the generator suppressed (this is the default state in VS)
+            driver = SetSuppressionState(true);
+
+            // results should be empty, and no recorded steps should have run
+            using var eventListener = new RazorEventListener();
+            var result = RunGenerator(compilation!, ref driver).VerifyPageOutput();
+            Assert.DoesNotContain("DocumentsWithSuppression", (IReadOnlyDictionary<string, ImmutableArray<IncrementalGeneratorRunStep>>)result.TrackedSteps);
+            Assert.DoesNotContain("ImplementationSourceOutput", (IReadOnlyDictionary<string, ImmutableArray<IncrementalGeneratorRunStep>>)result.TrackedOutputSteps);
+            Assert.Empty(eventListener.Events);
+
+            // Now enable the generator and confirm we get the expected output
+            driver = SetSuppressionState(false);
```

Comment:
From the linked issue I understand this happens during hot reload. Can you share more details when do we suppress and unsuppress the generator there?

Reply:
By default, in Roslyn the suppression state flag is set. If you never do a hot reload it stays that way, and so the generator never runs. When you do a hot reload/EnC, it modifies the workspace to remove the flag, meaning the generator actually runs. 

I had thought that because EnC forks the workspace it stays that way, and never turns it back off again, but it does seem to (hence this bug). So in the hot reload workspace the flag can essentially toggle on and off arbitrarily. This PR makes sure that when that happens we keep the caches full so the next 'on' phase doesn't have to re-create everything from scratch.